src/app/image-group-page/image-group-page.component.ts

```diff
@@ -324,33 +303,50 @@ export class ImageGroupPage implements OnInit, OnDestroy {
     }
   }
 
-  private exportGroup(event: any): void {
+  private exportGroup(exportType: SupportedExportTypes): void {
     this.showTermsConditions = false
 
-    switch (event) {
-      case 'ppt': {
-        // Perform PPT download action
-        this.getPPT()
-        break
-      }
-      case 'GoogleSlides': {
-        if(this._storage.getSession('GAuthed')) {
-          // Export to GS and show loading state
-          console.log('Export to GS and show loading state')
-        } else {
-          this.showGoogleAuth = true
+    this._ig.getImageIdsToDownload(this.ig)
+      .then((imageIdsToDownload) => {
+        // If user has agreed, we should trigger download directly
+        switch (exportType) {
+          case SupportedExportTypes.PPTX: {
+            // Perform PPT download action
+            this.executeBulkExport(imageIdsToDownload, SupportedExportTypes.PPTX)
+            break
+          }
+          case 'GoogleSlides': {
+            if(this._storage.getSession('GAuthed')) {
+              // Export to GS and show loading state
+              console.log('Export to GS and show loading state')
+            } else {
+              this.showGoogleAuth = true
+            }
+            break
+          }
+          case SupportedExportTypes.ZIP: {
+            // Perform ZIP download action
+            this.executeBulkExport(imageIdsToDownload, SupportedExportTypes.ZIP)
+            break
+          }
```

Comment:
This case could be combined with PPTX and the second arg can become the switch condition. 
```
this.executeBulkExport(imageIdsToDownload, exportType)
```